### ISSUE TITLE
xdoc token columns are unclear and possibly missing information the user needs

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2454]
We have 2 columns of information on tokens in xdocs, but it is based on 3 fields (acceptable, required, default).
It is unclear how the column are built from the 3 fields and may need to change?
example: the columns could be direct copies of acceptable/default, or could be acceptable/default minus required tokens

Does the user need to know this information, or just enough to customize the check to their needs?

Currently if they copy the tokens to their own config, it will crash because we don't display required tokens, which they must input.
But we don't display the tokens if acceptable = required = default, but in this case they can't do any customization on tokens anyways. The only issue this causes is we don't inform them which tokens the check is using.

**Comment by rnveach** [comment]
my opinion:
We should always append "required" tokens to the user's config input. They are literally required. The user can't run without them no matter what. They should only be allowed to specify 'acceptable - required'.
The documentation should then show only these properties the user can add/remove (acceptable/default minus required).
We should then add a new row to the "Properties" table called "required tokens", that informs them what those tokens are. But when will the user need to know this?

**Comment by MEZk** [comment]
@rnveach 
You're right, It's no use to specify acceptable and required tokens in xdocs when required tokens fully includes default tokens and/or acceptable tokens. User does not have an opportunity to configure tokens for such checks.

+1 for adding required tokens as a column in xdocs.
